### PR TITLE
add latest sol2 version 2 release

### PR DESCRIPTION
**Pull request by a4z**
This is for people who stuck to a pre c++17 compiler

Specify library name and version:  **sol2/2.20.6**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by a4z** [comment]
please help!
I don't understand the problem. Is it not possible to add older versions?

**Comment by a4z** [comment]
please ack that one
https://github.com/conan-io/conan-center-index/pull/1357
so I can continue with this one. Thanks

**Comment by SpaceIm** [comment]
It's a bug in CCI (the same than https://github.com/conan-io/conan-center-index/pull/1312). Pretty sure that https://github.com/conan-io/conan-center-index/pull/1357 won't help and has 0 chance to accepted 😛

File: recipes/sol2/config.yml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 versions:
+  2.20.6:
+    folder: 2.x.x
   3.0.3:
     folder: 3.x.x
```
Comment (uilianries) [inline recipes/sol2/config.yml:5]:
```suggestion
versions:
  "2.20.6":
    folder: "2.x.x"
  "3.0.3":
    folder: "3.x.x"
```

Comment (uilianries) [reply recipes/sol2/config.yml:5]: Try passing them to strings. I remember in the past we had a similar problem, that's why we use string in most of config.yml

Comment (SpaceIm) [reply recipes/sol2/config.yml:5]: PROJ recipe uses strings and has the same issue.

**Comment by a4z** [comment]
> It's a bug in CCI (the same than #1312). Pretty sure that #1357 won't help and has 0 chance to be accepted 😛

I think we still need 1357 , or have more complex logic in the recipe for the c++ version
the point of 2.x is to support compilers pre c++17 

PS: the change with the quotes did not help

**Comment by a4z** [comment]
I close this PR and create a new one since this CI/CD bot problem seems to be not solvable even after merging master, ....

**Comment by a4z** [comment]
tried a new pr, its the same problem :-( 
https://github.com/conan-io/conan-center-index/pull/1385

**Pull request closed without merging**